dbt_coverage/__init__.py

```diff
@@ -75,6 +97,37 @@ class Catalog:
 
     tables: Dict[str, Table]
 
+    def filter_catalog(self, model_path_filter: List[str]) -> Catalog:
+        """
+        Filter ``Catalog``'s ``tables`` attribute to ``Tables`` that have a ``model_path_filter``
+        value as part of their ``original_file_path``.
+
+        :param model_path_filter: the model_path string(s) to filter tables on, (matches using
+        the ``in`` operator)
+
+        :returns: Catalog
+        :raises ValueError: if no ``Table`` in the ``tables`` Catalog attribute have an
+        ``original_file_path`` that contains any ``model_path_filter`` value
+        """
+        filtered_tables = {}
+
+        original_tables_dict = {key: val for key, val in self.tables.items()}
+        for key, table in original_tables_dict.items():
+            for path in model_path_filter:
+                if path in table.original_file_path:
```

Comment:
@followingell how would you feel about changing this to `table.original_file_path.startswith(path)`? I am mostly concerned about the unintended consequences of having say the `staging/` and `pre-staging/` folders in one's project and as it currently is, I am afraid `--model-path-filter staging/` would cover both.

Alternatively, having a regex option would certainly work too 🙂

Reply:
@mrshu Again, great thoughts r.e. covering both `staging/` and `pre-staging/` unintentionally when using `--model-path-filter staging/` with my previous implementation.

As per your suggestion I have swapped the code to use `startswith` and had this reflected in the README since this provides as much control as before whilst reducing the potential for unintended inclusions as outlined above.